metrics/firebase/functions/lib/main.dart

```diff
@@ -42,7 +47,78 @@ Future<void> onBuildAddedHandler(DocumentSnapshot snapshot, _) async {
     statusFields: [statusFieldIncrement],
   );
 
-  await _updateBuildDay(snapshot, buildDayData);
+  final dataJson = Map<String, dynamic>.from(buildData.toJson());
+  dataJson.addAll({'id': buildId});
+
+  final onErrorData = TaskData(
+    code: TaskCode.buildDaysCreated,
+    data: dataJson,
+    createdAt: clock.now(),
+  );
+  final firestore = snapshot.firestore;
+
+  await _updateBuildDay(
+    firestore,
+    buildDayData: buildDayData,
+    onErrorData: onErrorData,
+  );
+}
+
+/// Process incrementing and decrementing logic for the build days collection
+/// document, based on a updated build's status and started date.
+Future<void> onBuildUpdatedHandler(Change<DocumentSnapshot> change, _) async {
+  final oldBuild = BuildDataDeserializer.fromJson(change.before.data.toMap());
+  final newBuild = BuildDataDeserializer.fromJson(change.after.data.toMap());
+  final startedAtDayUtc = _getUtcDay(newBuild.startedAt);
+  final buildId = '${newBuild.projectId}_${newBuild.buildNumber}';
+  final firestore = change.before.firestore;
+  final statusFields = <BuildDayStatusField>[];
+
+  final task = await _getTaskByDataId(firestore, buildId);
+
+  if (task == null) {
+    final statusFieldDecrement = _createBuildDayStatusField(
```

Comment:
Should we handle the situation when we'll have a task with the `build_days_updated` code?